src/cljx/schema/core.cljx

```diff
@@ -522,17 +495,16 @@
 
 (clojure.core/defrecord ConditionalSchema [preds-and-schemas]
   Schema
-  (walker [this]
-          (let [preds-and-walkers (mapv (clojure.core/fn [[pred schema]] [pred (subschema-walker schema)])
-                                        preds-and-schemas)]
-            (clojure.core/fn [x]
-              (if-let [[_ match] (first (filter (clojure.core/fn [[pred]] (pred x)) preds-and-walkers))]
-                (match x)
-                (macros/validation-error this x (list 'matches-some-condition? (utils/value-name x)))))))
+  (spec [this]
+    (variant/variant-spec
+     spec/+no-precondition+
+     (for [[p s] preds-and-schemas]
+       {:guard p :schema s})
+     #(list 'matches-some-condition? %))) ;; TODO: improve error message
```

Comment:
:sheep: how about `no-matching-condition?` to mirror the error message from clojure's `case`: `IllegalArgumentException No matching clause`